pkg/util/limiter/query_limiter.go

```diff
@@ -15,25 +16,32 @@ import (
 type queryLimiterCtxKey struct{}
 
 var (
-	ctxKey          = &queryLimiterCtxKey{}
-	errMaxSeriesHit = "The query hit the max number of series limit (limit: %d)"
+	ctxKey              = &queryLimiterCtxKey{}
+	errMaxSeriesHit     = "The query hit the max number of series limit (limit: %d)"
+	ErrMaxChunkBytesHit = "The query hit the aggregated chunks size limit (limit: %d bytes)"
 )
 
 type QueryLimiter struct {
 	uniqueSeriesMx sync.Mutex
 	uniqueSeries   map[model.Fingerprint]struct{}
 
-	maxSeriesPerQuery int
+	chunkBytesCount *atomic.Int64
```

Comment:
nit: When embedding atomic into other structs, and these structs are only accessed via pointers, atomic itself doesn't need to be a pointer.

In other words, `chunkBytesCount atomic.Int64` would work fine here, and then there is no need to call `atomic.NewInt64(0)` in `NewQueryLimiter`.